src/main/api/Abilities.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * MacroQuest: The extension platform for EverQuest
+ * Copyright (C) 2002-2023 MacroQuest Authors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "pch.h"
+
+#include "mq/api/Abilities.h"
+
+#include "eqlib/PcClient.h"
+#include "eqlib/EQClasses.h"
+
+// Defined in MQ2Globals.cpp but we don't want to bring in all of MQ2Globals.h
+MQLIB_VAR const char* szInnates[];
+
+namespace mq {
+
+int GetAdjustedSkill(int nSkill)
+{
+	return pLocalPC ? pLocalPC->GetAdjustedSkill(nSkill) : 0;
+}
+
+int GetBaseSkill(int nSkill)
+{
+	return pLocalPC ? pLocalPC->GetBaseSkill(nSkill) : 0;
+}
+
+bool HasSkill(int nSkill)
+{
+	return pLocalPC && pLocalPC->HasSkill(nSkill);
+}
+
+bool SkillIsActivatable(int nSkill)
+{
+	return nSkill < NUM_SKILLS && pSkillMgr && pSkillMgr->pSkill[nSkill]->Activated;
+}
+
+bool HasInnateSkill(int nSkill)
+{
+	// Innate skills stack on top of normal skills when used with Skill Manager
+	// Allow for both
+	if (nSkill >= NUM_SKILLS)
+		nSkill -= NUM_SKILLS;
+
+	if (nSkill < 0 || nSkill > NUM_INNATE)
+		return false;
+
+	const PcProfile* pProfile = pLocalPC ? pLocalPC->GetCurrentPcProfile() : nullptr;
+	// 255 is the default value for an unset innate skill
+	return 	pProfile && pProfile->InnateSkill[nSkill] != 255;
```

Comment:
PcProfile is always going to exist if pLocalPC exists, so this could be simplified.

Reply:
```cpp
	if (pLocalPC == nullptr)
		return false;
//snip
	return pLocalPC->GetCurrentPcProfile()->InnateSkill[nSkill] != 255;
```
?

Reply:
or

```
return pLocalPC && pLocalPC->GetCurrentPcProfile()->InnateSkill[nSkill] != 255;
```

although part of me wants to add an inline accessor to PcClinet to simplify this further like...

`pLocalPC->GetInnateSkill(nSkill) != 255`

Reply:
I worry about the parallels in AdjustedSkill :D